src/org/javarosa/core/model/utils/DateUtils.java

```diff
@@ -69,6 +68,23 @@ public DateFields () {
 //            tzOffset = 0;
         }
 
+        public DateFields(int year, int month, int day, int hour, int minute, int second, int secTicks) {
+            this.year = year;
+            this.month = month;
+            this.day = day;
+            this.hour = hour;
+            this.minute = minute;
+            this.second = second;
+            this.secTicks = secTicks;
+            // The official API returns a day of week [1=monday, 7=sunday].
+            // We normalize to current JR standard [0=sunday, 6 saturday] by getting the module of 6
+            this.dow = LocalDateTime.of(year, month, day, hour, minute, second, secTicks * 1_000_000).getDayOfWeek().getValue() % 6;
```

Comment:
Maybe this would be clearer than the modulo operation:
```
int iso8601Dow = LocalDateTime.of ... .getValue()
this.dow = iso8601Dow == 7 ? 0 /* JR uses 0 for Sunday */ : iso8601Dow
```